### PR TITLE
Extend migration tests

### PR DESCRIPTION
**Pull request by farost**
## Usage and product changes
Extend the migration tests for the Rust driver to include more corner cases of incomplete database imports.

## Implementation
Update refs.
Add new steps for the Rust driver -- it's the only driver that can manually control the import stream to emulate the steps required. 
Extend the C# test infrastructure to allow it to parse ignored scenarios.

### CODE REVIEW

File: .factory/automation.yml [hunk 1]
```diff
@@ -156,14 +156,8 @@ build:
...
-          bazel test --config=ci //rust/tests/behaviour/driver:test_concept --test_env=ROOT_CA=$ROOT_CA --//rust/tests/behaviour/config:mode=cluster --test_output=streamed &&
-          bazel test --config=ci //rust/tests/behaviour/driver:test_connection --test_env=ROOT_CA=$ROOT_CA --//rust/tests/behaviour/config:mode=cluster --test_output=streamed &&
-          bazel test --config=ci //rust/tests/behaviour/driver:test_query --test_env=ROOT_CA=$ROOT_CA --//rust/tests/behaviour/config:mode=cluster --test_output=streamed &&
-          bazel test --config=ci //rust/tests/behaviour/driver:test_user --test_env=ROOT_CA=$ROOT_CA --//rust/tests/behaviour/config:mode=cluster --test_output=streamed &&
-          bazel test --config=ci //rust/tests/behaviour/driver:test_cluster --test_env=ROOT_CA=$ROOT_CA --//rust/tests/behaviour/config:mode=cluster --test_output=streamed &&
+          bazel test --config=ci //rust/tests/behaviour/... --test_env=ROOT_CA=$ROOT_CA --//rust/tests/behaviour/config:mode=cluster --test_output=streamed &&
```
Comment (farost) [inline .factory/automation.yml:160]: We can run all the tests together safely now

File: csharp/Test/Behaviour/Driver/Migration/MigrationTest.cs [hunk 2]
```diff
@@ -21,7 +21,7 @@
 
 namespace TypeDB.Driver.Test.Behaviour
 {
-    [FeatureFile("external/typedb_behaviour+/driver/migration.feature")]
+    [FeatureFile("migration.csharp.feature")]
```
Comment (farost) [inline csharp/Test/Behaviour/Driver/Migration/MigrationTest.cs:24]: We generate a new file without ignored cases. Only needed for migration

File: csharp/Test/Behaviour/Util/FeatureFileFilter.cs
Comment (farost) [inline csharp/Test/Behaviour/Util/FeatureFileFilter.cs:1]: A filter for ignored BDD scenarios. The library doesn't provide anything better, it's quite limited

File: rust/src/connection/database/import_stream.rs [hunk 3]
```diff
@@ -27,7 +27,8 @@ use crate::{
     promisify, resolve,
 };
 
-pub(crate) struct DatabaseImportStream {
+#[doc(hidden)]
+pub struct DatabaseImportStream {
```
Comment (farost) [inline rust/src/connection/database/import_stream.rs:31]: I want to add more BDD tests for import, which requires exposing these classes as `pub`. I at least hide the docs. I don't think it's crucial.

**Review by dmitrii-ubskii — APPROVED** [review]
